pkg/network/infraconfigurators/bridge_test.go

```diff
@@ -634,6 +635,81 @@ var _ = Describe("Bridge infrastructure configurator", func() {
 			})
 		})
 	})
+
+	Context("DHCP configuration generation", func() {
+		const (
+			ifaceName   = "eth0"
+			launcherPID = 1000
+		)
+
+		var (
+			bridgeConfigurator *BridgePodNetworkConfigurator
+			defaultGwRoute     netlink.Route
+			iface              *v1.Interface
+			podLink            *netlink.GenericLink
+			podIP              netlink.Addr
+			vmi                *v1.VirtualMachineInstance
+			mac                net.HardwareAddr
+			mtu                int
+		)
+
+		BeforeEach(func() {
+			iface = v1.DefaultBridgeNetworkInterface()
+			vmi = newVMIBridgeInterface("default", "vm1")
+
+			bridgeConfigurator = NewBridgePodNetworkConfigurator(vmi, iface, bridgeIfaceName, launcherPID, handler)
+		})
+
+		When("IPAM is not enabled", func() {
+			BeforeEach(func() {
+				mtu = 1000
+				podLink = &netlink.GenericLink{LinkAttrs: netlink.LinkAttrs{Name: ifaceName, MTU: mtu}}
+				bridgeConfigurator.podNicLink = podLink
+				macStr := "AF:B3:1F:78:2A:CA"
+				mac, _ = net.ParseMAC(macStr)
+
+			})
+
+			It("should generate a minimal dhcp configuration", func() {
+				expectedDhcpConfig := cache.DHCPConfig{IPAMDisabled: true}
+				Expect(bridgeConfigurator.GenerateNonRecoverableDHCPConfig()).To(Equal(&expectedDhcpConfig))
+			})
+		})
+
+		When("IPAM is enabled", func() {
+			BeforeEach(func() {
+				podLink = &netlink.GenericLink{LinkAttrs: netlink.LinkAttrs{Name: ifaceName}}
+				podIP = netlink.Addr{IPNet: &net.IPNet{IP: net.IPv4(10, 35, 0, 6), Mask: net.CIDRMask(24, 32)}}
+			})
+
+			BeforeEach(func() {
+				bridgeConfigurator.podNicLink = podLink
+				bridgeConfigurator.podIfaceIP = podIP
+				bridgeConfigurator.ipamEnabled = true
+				bridgeConfigurator.vmMac = &mac
+			})
+
+			When("routes are installed", func() {
```

Comment:
Don't we need to add a test to IPAM and routes missing ?

Reply:
Good catch; done.